### PR TITLE
Reproducing the test case where nested set of conditionals point to t…

### PR DESCRIPTION
**Pull request by dvargas92495**
…he wrong final output nodes

Discovered a variant of this graph on a customer workflow. I tried to MVP it as much as possible, but without going into solution mode, couldn't break it down further.

I believe the issue is once we create a set of nodes as an edge target, we need to break the set if we discover an edge pointing to a member of that set.

### CODE REVIEW

File: ee/codegen/src/__test__/graph-attribute.test.ts [hunk 1]
```diff
@@ -686,5 +687,99 @@ describe("Workflow", () => {
...
+        [entrypointNode, firstCheckNode],
+        [[firstCheckNode, "if_port"], firstInnerCheckNode],
+        [[firstCheckNode, "else_port"], finalCheckNode],
+        [[finalCheckNode, "if_port"], innerTerminalNode],
+        [[finalCheckNode, "else_port"], outerOutputNode],
+        [[secondInnerCheckNode, "if_port"], finalCheckNode],
+        [[secondInnerCheckNode, "else_port"], outerOutputNode],
+        [[firstInnerCheckNode, "if_port"], secondInnerCheckNode],
+        [[firstInnerCheckNode, "else_port"], finalCheckNode],
+      ]);
```
Comment (dvargasfuertes) [inline ee/codegen/src/__test__/graph-attribute.test.ts:758]: This graph is why I had all those PRs cleaning up the graph attribute tests. I anticipate way more complex edge cases to come

File: ee/codegen/src/__test__/__snapshots__/graph-attribute.test.ts.snap [hunk 2]
```diff
@@ -174,6 +174,25 @@ exports[`Workflow > graph > should be correct for two branches merging from sets
 "
 `;
 
+exports[`Workflow > graph > should be pointing to the correct terminal nodes from a nested set of conditionals 1`] = `
```
Comment (dvargasfuertes) [inline ee/codegen/src/__test__/__snapshots__/graph-attribute.test.ts.snap:177]: Manually edited this snapshot to match the intended target

Comment (dvargasfuertes) [reply ee/codegen/src/__test__/__snapshots__/graph-attribute.test.ts.snap:177]: Note that there are actually three valid graphs possible. The set for `FinalCheckNode` could go in any of the three spots where its referenced

**Review by marlenc3 — APPROVED** [review]
